lib/runner.js

```diff
@@ -44,67 +43,76 @@ Runner.prototype.addAfterHook = function (title, cb) {
 	this.tests.after.push(new Test(title, cb));
 };
 
-Runner.prototype.concurrent = function (tests, cb) {
-	each(tests, function (test, i, next) {
-		test.run(function (err, duration) {
-			if (err) {
-				this.stats.failCount++;
-			}
-
-			this.results.push({
-				duration: duration,
-				title: test.title,
-				error: err
-			});
+Runner.prototype.concurrent = function (tests) {
+	// run all tests
+	tests = tests.map(function (test) {
+		return test.run();
+	});
+
+	return Promise.all(tests, function (test) {
+		return test
+			.catch(function () {
+				return;
+			})
+			.then(function () {
+				this.addTestResult(test);
+			}.bind(this));
+	}.bind(this));
+};
 
-			this.emit('test', err, test.title, duration);
-			next();
-		}.bind(this));
-	}.bind(this), cb);
+Runner.prototype.serial = function (tests) {
+	return Promise.resolve(tests).each(function (test) {
+		return test.run()
+			.catch(function () {
+				return;
+			})
+			.then(function () {
+				this.addTestResult(test);
+			}.bind(this));
+	}.bind(this));
 };
 
-Runner.prototype.serial = function (tests, cb) {
-	eachSerial(tests, function (test, next) {
-		test.run(function (err, duration) {
-			if (err) {
-				this.stats.failCount++;
-			}
+Runner.prototype.addTestResult = function (test) {
```

Comment:
This should be a private method, meaning prefixed with `_`: `Runner.prototype._addTestResult`